API/application/iotjs.py

```diff
@@ -163,51 +158,52 @@ def build(self):
             # Enable memstat for IoT.js (libtuv, jerryscript, iotjs)
             self.apply_patches()
 
-        elif self.device.get_type() == 'rpi2' and self.os == 'linux':
-            build_flags.append('--target-board=%s' % self.device.get_type())
+        elif device.get_type() == 'rpi2':
+            build_flags.append('--target-board=%s' % device.get_type())
             build_flags.append('--jerry-cmake-param=-DFEATURE_VALGRIND_FREYA=ON')
             build_flags.append('--compile-flag=-g')
             build_flags.append('--jerry-compile-flag=-g')
 
-        elif self.device.get_type() == 'artik053' and self.os == 'tizenrt':
+        elif device.get_type() == 'artik053':
             build_flags.append('--target-board=artik05x')
             build_flags.append('--sysroot=%s' % paths.TIZENRT_OS_PATH)
 
         else:
             console.fail('Non-minimal IoT.js build failed, unsupported '
-                         'device-os (%s-%s) combination!' %
-                         (self.device.get_type(), self.os))
+                         'device (%s)!' % device.get_type())
```

Comment:
That was created to throw an error when a not supported device - os combination was defined by the user (e.g. rpi2 - nuttx). Since this patch hides the `os` selection from the user, we can remove the `else` case fully.

Reply:
I think we should keep this case. It is a remainder for us which ensures to set the appropriate build flags when adding a new device.